File: sdk/go/bot_config.go
package openwa

import "context"

// BotConfigService is per-session bot config (access lists, prefix, commands, welcome).
// Backed by src/modules/bot/bot-config.controller.ts.
type BotConfigService struct{ client *Client }

func (s *BotConfigService) path(sessionID string) string {
	return "/api/sessions/" + pathEscape(sessionID) + "/bot-config"
}

func (s *BotConfigService) Get(ctx context.Context, sessionID string) (map[string]any, error) {
	var out map[string]any
	err := s.client.do(ctx, "GET", s.path(sessionID), nil, nil, &out)
	return out, err
}

func (s *BotConfigService) Update(ctx context.Context, sessionID string, body map[string]any) (map[string]any, error) {
	var out map[string]any
	err := s.client.do(ctx, "PUT", s.path(sessionID), nil, body, &out)
	if err != nil {
		return nil, err
	}
	return out, nil
}

File: sdk/javascript/src/resources/bot-config.ts
/**
 * Per-session bot config — access lists, prefix, commands, autoRead, welcome.
 *
 * Backed by `src/modules/bot/bot-config.controller.ts`.
 * @packageDocumentation
 */

import { encodeSegment } from '../http.js';
import type { OpenWAClient } from '../client.js';

export interface BotConfigRecord {
  sessionId: string;
  accessMode: 'all' | 'allow' | 'block';
  allowList: string[];
  blockList: string[];
  prefix: string;
  commandsEnabled: boolean;
  autoRead: boolean;
  alwaysOnline: boolean;
  welcomeMessage: string | null;
  createdAt: string;
  updatedAt: string;
}

export type UpdateBotConfigRequest = Partial<
  Omit<BotConfigRecord, 'sessionId' | 'createdAt' | 'updatedAt'>
>;

export class BotConfigResource {
  constructor(private readonly client: OpenWAClient) {}

  get(sessionId: string): Promise<BotConfigRecord> {
    return this.client.request({
      method: 'GET',
      path: `/api/sessions/${encodeSegment(sessionId)}/bot-config`,
    });
  }

  update(sessionId: string, body: UpdateBotConfigRequest): Promise<BotConfigRecord> {
    return this.client.request({
      method: 'PUT',
      path: `/api/sessions/${encodeSegment(sessionId)}/bot-config`,
      body,
    });
  }
}
